refinery/core/views.py

```diff
@@ -925,37 +908,73 @@ def delete(self, request, uuid):
 
 class GroupViewSet(viewsets.ViewSet):
     """API endpoint for viewing groups."""
-    http_method_names = ['get']
+    http_method_names = ['get', 'patch']
+    lookup_field = 'uuid'
 
-    def list(self, request):
-        data_set_uuid = request.query_params.get('dataSetUuid')
+    def get_object(self, uuid):
         try:
-            data_set = DataSet.objects.get(uuid=data_set_uuid)
-        except DataSet.DoesNotExist as e:
+            return ExtendedGroup.objects.get(uuid=uuid)
+        except ExtendedGroup.DoesNotExist as e:
             logger.error(e)
-            return HttpResponseNotFound(
-                 content="DataSet with UUID: {} not found.".format(
-                     data_set_uuid
-                 )
-             )
-        except DataSet.MultipleObjectsReturned as e:
+            raise Http404
+        except ExtendedGroup.MultipleObjectsReturned as e:
             logger.error(e)
-            return HttpResponseServerError(
-                content="Multiple dataSets returned for this request"
-            )
+            raise APIException("Multiple groups returned for this request.")
+
+    def list(self, request):
+        data_set_uuid = request.query_params.get('dataSetUuid')
+        all_perms_flag = request.query_params.get('allPerms', False)
+
+        data_set = get_data_set_for_view_set(data_set_uuid)
 
         public_group = ExtendedGroup.objects.public_group()
         if not ('read_meta_dataset' in get_perms(public_group, data_set) or
                 request.user.has_perm('core.read_meta_dataset', data_set)):
-            return Response(data_set_uuid, status=status.HTTP_401_UNAUTHORIZED)
+            return Response(data_set_uuid, status=status.HTTP_403_FORBIDDEN)
```

Comment:
Why not raise PermissionDenied?
https://www.django-rest-framework.org/api-guide/exceptions/#permissiondenied